fastestimator/trace/metric/precision.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import numpy as np
+from sklearn.metrics import precision_score
+from typing import Optional
+
+from fastestimator.backend.to_number import to_number
+from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
+
+
+class Precision(Trace):
+    """Computes precision for classification task and report it back to logger.
+
+    Args:
+        true_key (str): Name of the keys in the ground truth label in data pipeline.
```

Comment:
we're going to type hint in code instead of doc string now

Reply:
removed arguments type in the docstring